lib/ecto/soft_delete_repo.ex

```diff
@@ -26,6 +26,21 @@ defmodule Ecto.SoftDelete.Repo do
   """
   @callback soft_delete_all(queryable :: Ecto.Queryable.t()) :: {integer, nil | [term]}
 
+  @doc """
+  Soft restores all entries matching the given query.
+
+  It returns a tuple containing the number of entries and any returned
+  result as second element. The second element is `nil` by default
+  unless a `select` is supplied in the update query.
+
+  ## Examples
+
+    MyRepo.soft_restore_all(%Post{}, MyRepo)
+
+
+  """
+  @callback soft_restore_all(struct :: Ecto.Schema.t()) :: {integer, nil | [term]}
```

Comment:
```suggestion
  @callback soft_restore_all(struct :: Ecto.Queryable.t()) :: {integer, nil | [term]}
```

Queryable are: `structs`, `schema`, `ecto query` (maybe I miss some queryable element)

because you can write 

```elixir
from(entity in "my_table", where entity.some_field = "some_value") 
|> soft_restore_all()
```